biz.aQute.bnd.maven/src/aQute/bnd/maven/lib/resolve/BndrunContainer.java

```diff
@@ -209,6 +215,21 @@ public FileSetRepository getFileSetRepository() throws Exception {
 		return fileSetRepository = dependencyResolver.getFileSetRepository(name, bundles, useMavenDependencies);
 	}
 
+	private void setEEfromBuild(Bndrun run) {
+		String runee = run.getProperty(Constants.RUNEE);
+
+		if (runee == null) {
+			Optional.ofNullable(project.getPlugin("org.apache.maven.plugins:maven-compiler-plugin"))
+				.map(Plugin::getConfiguration)
+				.map(Xpp3Dom.class::cast)
+				.map(dom -> dom.getChild("target"))
+				.map(Xpp3Dom::getValue)
+				.map(EE::highestFromTargetVersion)
```

Comment:
This needs to be a `flatMap` now since the method returns an Optional. Then you don't need the `.map(Optional::get)` :-)

Reply:
Did your recent push force intend to address the `flatMap` comment above?